src/app/wallets/get-transactions-for-wallet.ts

```diff
@@ -42,8 +49,12 @@ export const getTransactionsForWallets = async (
 
   // we are getting both the transactions in the mempool and the transaction that
   // have been mined by not yet credited because they haven't reached enough confirmations
-  const onChainTxs = await onChain.listIncomingTransactions(ONCHAIN_MIN_CONFIRMATIONS)
-  if (onChainTxs instanceof OnChainError) {
+  const onChainTxs = await redisCache.getOrSet({
+    key: CacheKeys.LastOnChainTransactions,
+    ttlSecs: SECS_PER_10_MINS,
```

Comment:
In theory this shouldn't matter because a partial transactions list is built from the ledger before this code and is early returned if the `OnChainService` errors which would likely happen before this code when we're instantiating the `OnChainService`. This TTL would only affect pending incoming txs I believe, so it's not the same issue we've been observing.

Longer TTL hedges us against a very small window where the `OnChainService` successfully comes up and then a `listIncomingTransactions` call fails. Tradeoff is new pending incoming txs are more delayed in showing up in the API.

If these assumptions are all correct, I think shorter TTL is ok here.

Reply:
ok yes thanks for the input. I'm approving it.